app/services/legacy/dataset_import_service.rb

```diff
@@ -35,49 +35,79 @@ def update_or_create_dataset
-    secondary_theme  = legacy_dataset["theme-secondary"]
+    old_theme = legacy_dataset["theme-primary"]
+    secondary_theme = legacy_dataset["theme-secondary"]
     dataset.theme_id = themes_cache.fetch(old_theme, nil)
     dataset.secondary_theme_id = themes_cache.fetch(secondary_theme, nil)
     dataset.status = "published"
     dataset.save!(validate: false)
   end
 
   def create_datafiles(dataset)
-    resources.each do |resource|
-      create_resource(resource, dataset)
+    create_timeseries_datafiles(dataset)
+    create_non_timeseries_datafiles(dataset)
+    create_additional_info_datafiles(dataset)
+  end
+
+  def create_additional_info_datafiles(dataset)
+    Array(@legacy_dataset['additional_resources']).each do |resource|
+      datafile = Doc.find_or_create_by(url: resource["url"], dataset_id: dataset.id)
+      base_attributes = create_datafile_base_attributes(resource, dataset)
+
+      datafile.assign_attributes(base_attributes)
+      datafile.save!(validate: false)
+    end
+  end
+
+  def create_non_timeseries_datafiles(dataset)
+    Array(@legacy_dataset['individual_resources']).each do |resource|
+      datafile = Doc.find_or_create_by(url: resource["url"], dataset_id: dataset.id)
+      base_attributes = create_datafile_base_attributes(resource, dataset)
+
+      datafile.assign_attributes(base_attributes)
+      datafile.save!(validate: false)
     end
   end
 
-  def create_resource(resource, dataset)
-    datafile = find_or_initialize(resource)
-    datafile.uuid = resource["id"]
-    datafile.format = resource["format"]
-    datafile.name = resource["description"].presence || "No name specified"
-    datafile.created_at = dataset.created_at
-    datafile.updated_at = dataset.last_updated_at
-
-    if resource["date"].present? && !documentation?(resource['format'])
-      dates = get_start_end_date(resource["date"])
-      if dataset.frequency != 'never'
-        begin
-          datafile.start_date = Date.parse(dates[0])
-        rescue ArgumentError
-          datafile.start_date = Date.new(1,1,1)
-        end
-        begin
-          datafile.end_date = Date.parse(dates[1])
-        rescue ArgumentError
-          datafile.end_date = Date.new(1,1,1)
-        end
-      end
+  def create_timeseries_datafiles(dataset)
+    Array(@legacy_dataset['timeseries_resources']).each do |resource|
+      datafile = Link.find_or_create_by(url: resource["url"], dataset_id: dataset.id)
+      base_attributes = create_datafile_base_attributes(resource, dataset)
+      date_attributes = create_datafile_date_attributes(resource)
+
+      datafile.assign_attributes(base_attributes)
+      datafile.assign_attributes(date_attributes)
+
+      datafile.save!(validate: false)
     end
+  end
+
+  def create_datafile_base_attributes(resource, dataset)
+    {
+      uuid: resource["id"],
+      format: resource["format"],
+      name: datafile_name(resource),
+      created_at: dataset.created_at,
```

Comment:
Isn't `created_at` a default set by active record when you create a datafile. Not sure if it's possible/wise to try and overwrite it??